linkerd/app/inbound/src/lib.rs

```diff
@@ -283,15 +285,21 @@ impl Config {
                 http_server.into_inner(),
                 h2_settings,
                 drain.clone(),
-                disable_protocol_detection_for_ports.clone(),
             );
 
-            // Terminate inbound mTLS from other outbound proxies.
-            let accept = tls::AcceptTls::new(local_identity, tcp_server)
-                .with_skip_ports(disable_protocol_detection_for_ports);
+            let tcp_detect = svc::stack(tcp_server)
+                .push(DetectProtocolLayer::new(ProtocolDetect::new(
+                    disable_protocol_detection_for_ports.clone(),
+                )))
```

Comment:
Nit, take it or leave it: I find it a _little_ weird we have to constuct a `DetectProtocolLayer` _and_ a `ProtocolDetect` here. Does it make sense to have one constructor (maybe for `DetectProtocolLayer`) that also builds the `ProtocolDetect`?

Reply:
IIRC `ProtocolDetect` is a concrete strategy and the layer is generic.

Reply:
Yes, that is correct.